### PR TITLE
fix: fixes for setup issues

### PR DESCRIPTION
**Pull request by jedau**
# Fix: Contributor Setup Issues - ENOENT Video Errors and Node Version Update

## 📋 Description

This PR addresses multiple issues reported by contributors who forked the repository, focusing on video recording failures, missing Playwright browsers, and Node.js version compatibility. The changes improve error handling, documentation clarity, and debugging capabilities.

**Key Changes:**

1. **Enhanced Video Recording Error Handling** (`test/support/hooks.ts`)
   - Close page before getting video path to ensure video is finalized
   - Added `existsSync` check before attempting to read video files
   - Enhanced error logging with console.error for better visibility
   - Improved BeforeAll directory creation with try-catch error logging
   - Changed video attachment failure handling to return `pageClosedEarly: true`

2. **Simplified Test Execution** (`scripts/run-test.ts`)
   - Replaced async `exec()` with synchronous `execSync()` for more reliable process management
   - Used `stdio: 'inherit'` for direct output streaming
   - Eliminated complex error handling that caused platform-specific issues

3. **Node.js Version Upgrade**
   - Upgraded from v22.21.1 to v22.22.0 in `.nvmrc`
   - Updated version references in `docs/CONTRIBUTING.md` and `docs/TROUBLESHOOTING.md`

4. **Documentation Improvements**
   - Added comprehensive ENOENT troubleshooting section in `docs/TROUBLESHOOTING.md`
   - Made `npx playwright install` a **required** step in `docs/GETTING-STARTED.md` (was previously marked as optional)
   - Added bold warning about consequences of skipping browser installation
   - Added troubleshooting commands for both bash and PowerShell

5. **Debug Logging Configuration**
   - Changed `.env.example` DEBUG setting from `DEBUG=*` to `DEBUG=framework:*,test:*`
   - Reduces noise from verbose Playwright protocol logs (pw:browser, pw:api)

6. **Documentation Artifacts**
   - Added `BACKLOG.md` - Comprehensive project backlog with 60 tickets organized into milestones
   - Added `PR_DESCRIPTION.md` - Example PR description following template

## 🔗 Related Issue

Addresses contributor-reported issues:

- "Failed to attach video: ENOENT" errors when tests fail
- "Executable doesn't exist" errors due to missing Playwright browsers
- Platform-specific execution issues (Windows vs Mac/Linux)
- Contributors using incorrect Node.js version (v24.11.1 instead of v22.x)

## 🎯 Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🎨 Code style update (formatting, renaming)
- [x] ♻️ Refactoring (no functional changes)
- [ ] 🧪 Test update
- [x] 🔧 Configuration change
- [ ] 🚀 Performance improvement

## 🧪 Testing

- [x] All existing tests pass locally (Windows)
- [x] Video recording works correctly on test failure
- [x] BeforeAll creates directories with proper error handling
- [x] execSync properly streams output and exits cleanly
- [x] Documentation changes verified for accuracy

### Test Commands Run

```bash
# Verify setup from scratch
rm -rf node_modules
npm install
npx playwright install
npm test

# Test video recording on failure
npm test -- test/features/login.feature --tags "@negative"

# Verify all tests pass
npm test

# Generate and view report
npm run report
```

### Contributor Testing

Changes have been tested by multiple external contributors who reported the original issues.

## 📸 Screenshots (if applicable)

**Before - ENOENT Error:**

```
Failed to attach video: ENOENT: no such file or directory, open 'test-results/videos/...'
```

**After - Enhanced Error Message:**

```
Video file does not exist at: test-results/videos/abc123.webm
Failed to attach video: Error: ENOENT: no such file or directory
```

**Troubleshooting Guide - New Section:**

```markdown
### Error: "Failed to attach video: ENOENT"

**Solution:**

1. npx playwright install (most common cause)
2. mkdir -p test-results/videos test-results/traces test-results/screenshots
3. npm test

**Why this happens:**

- Fresh clones don't have test-results (gitignored)
- Tests fail before BeforeAll creates directories
- Playwright browsers not installed causes early failures
```

## ✅ Checklist

- [x] My code follows the project's naming conventions
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published
- [x] I have updated the README.md if needed
- [x] I have followed the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format for my commit messages

### Key Areas to Review

1. **run-test.ts Changes:**
   - Verify execSync with `stdio: 'inherit'` works across platforms
   - Confirm simplified implementation doesn't lose necessary error handling
   - Check that command arguments are properly escaped

2. **hooks.ts Video Recording:**
   - Validate page close → video path → file check sequence
   - Ensure existsSync check doesn't introduce race conditions
   - Review error logging approach (console.error vs other logging)

3. **Documentation Updates:**
   - Verify Node.js version references are consistent across all docs
   - Check that Playwright installation requirement is clear enough
   - Review ENOENT troubleshooting steps are accurate

4. **DEBUG Configuration:**
   - Confirm framework:_,test:_ captures all necessary logs
   - Verify it doesn't hide important error information
   - Consider if different defaults needed for development vs CI

### Testing Recommendations

**Manual Testing:**

```bash
# Fresh clone simulation
rm -rf node_modules test-results
npm install
# Skip playwright install to verify error handling
npm test  # Should fail with clear error message

# Full setup
npx playwright install
npm test  # Should run successfully
```

**Platform Testing:**

- ✅ Windows (verified by repo maintainer and contributors)
- ⚠️ macOS (needs verification)
- ⚠️ Linux (needs verification)

**Edge Cases:**

- Test failure triggering video recording
- Missing test-results directory on first run
- Playwright browsers not installed

### Why These Changes?

**Video Recording Issues:**

- Contributors frequently encountered ENOENT errors when tests failed
- Root cause was Playwright browsers not being installed (step was marked as optional)
- Video path was being accessed before the video file was ready
- No file existence check before attempting to read

**execSync vs exec:**

- exec() with callbacks had platform-specific behavior differences
- Windows could return string error codes, causing type errors
- execSync() with `stdio: 'inherit'` is simpler and more reliable
- Synchronous execution ensures proper process termination

**Node.js Version:**

- Contributors using various versions (v24.11.1, v22.21.1)
- Standardized on v22.22.0 for consistency
- Updated all documentation references

**DEBUG Configuration:**

- Default `DEBUG=*` showed excessive Playwright internals (pw:browser, pw:api, pw:channel)
- New setting `DEBUG=framework:*,test:*` focuses on relevant application/test logs
- Contributors can still enable full debugging when needed

### Migration Guide for Contributors

If you've already cloned the repository:

```bash
# Update Node.js version
nvm install v22.22.0
nvm use v22.22.0

# Update dependencies (if needed)
npm install

# Install/Update Playwright browsers
npx playwright install

# Update your .env file
# Change: DEBUG=*
# To: DEBUG=framework:*,test:*

# Run tests to verify
npm test
```

## 🔍 Reviewer Notes

**Please verify:**

1. **CI workflow changes** - Coverage job validation logic uses `@fail` tag instead of test name
2. **Error messages** - Enhanced troubleshooting guidance is helpful and actionable
3. **Documentation cleanup** - Duplicate Resources section removed from LEARNING-PATH.md
4. **Backward compatibility** - Changes don't break existing test workflows

**Testing verification:**

- CI coverage job should pass with expected output showing @fail test validation
- Regular test jobs continue to skip @fail tests (TAGS filter still in place)
- Error messages provide clear guidance if validation fails

**Review focus areas:**

- jq commands syntax in bash script
- Logic for detecting failed tests without @fail tag
- Error message clarity and actionability

---

**Before submitting this PR, please ensure:**

1. ✅ You have reviewed the [Branching Strategy](docs/BRANCHING-STRATEGY.md)
2. ✅ You have followed the [Pull Request Guidelines](docs/PULL-REQUEST.md)
3. ✅ Your branch is up to date with the base branch

### CODE REVIEW

File: test/support/hooks.ts [hunk 1]
```diff
@@ -88,14 +91,25 @@ async function saveVideo(page: Page, world: ICustomWorld): Promise<{ pageClosedE
...
+
+    // Read the video file
     const videoBuffer = await readFile(videoPath)
     world.attach(videoBuffer, 'video/webm')
     return { pageClosedEarly: true }
   } catch (error) {
-    console.warn('Failed to attach video:', error)
-    return { pageClosedEarly: false }
+    console.error('Failed to attach video:', error)
+    return { pageClosedEarly: true }
```
[the remark this answers is not in the document]
Comment (jedau) [reply test/support/hooks.ts]: Removed changes for this file for this PR

File: test/support/hooks.ts [hunk 2]
```diff
@@ -19,8 +19,11 @@ BeforeAll(async function () {
   const dirs = ['test-results/videos', 'test-results/traces', 'test-results/screenshots']
 
   for (const dir of dirs) {
-    if (!existsSync(dir)) {
+    try {
       await mkdir(dir, { recursive: true })
+    } catch (error) {
+      console.error(`Failed to create directory ${dir}:`, error)
+      throw error
     }
```
[the remark this answers is not in the document]
Comment (jedau) [reply test/support/hooks.ts]: Removed changes for this file for this PR
